Changelog updates (#20)
xataio/mdx-changelog: 2023-11-01.mdx

---
date: 11-01-2023
---

## What’s new?

Discover the latest enhancements and feature updates at Xata:

- **Added blog navigation by tags:** We've introduced a new feature that simplifies blog exploration. Now, you can easily discover blog posts by appending tags such as "postgres," "engineering," "product," "community," and more to the end of the blog tags URL (`https://xata.io/blog/tags`). This enables you to find and browse blog posts relevant to your specific areas of interest. For example, you can access all Xata blog posts related to PostgreSQL using the following URL: `https://xata.io/blog/tags/postgres`, or filter by two tags using a URL like this: `https://xata.io/blog/tags/community,announcements`. In addition, tags will soon be accessible directly through the UI for even easier navigation.

> - **Enhanced CSV file import support:** This update introduces the capability to import both remote and local CSV files, where only the file's path or URL is needed in the CSV file. While this feature is currently available in the TypeScript client, it is set to be expanded to the UI and CLI.

Reviewer: ```suggestion
- **Enhanced CSV file import support:** This update introduces the capability to import both remote and local files from a CSV. Only the file's path, URL or base64 encoding is required. While this feature is currently available in the CLI at the moment, we plan to support in the Xata app soon.
```

> - **Enhanced search responses:** This update introduces the `totalCount` field in search responses (including `search`, `search.byTable`, `search.all`, and `vectorSearch`) to report the total number of matching records. This change modifies the response format from an array to an object. The addition of this field provides more information about the search results, allowing users to understand the extent of data matching their search criteria.

Reviewer: ```suggestion
- **Enhanced search responses:** Closing a [common feature request](https://xata.canny.io/feature-requests/p/include-total-hits-in-search-response), the TypeScript SDK `v0.27.0` introduces the `totalCount` field in search responses (including `search`, `search.byTable`, `search.all`, and `vectorSearch`) to report the total number of matching records. This is considered a minor breaking change as the response format is now an object, not an array. Adding the total hits to every search response makes it easier for the client to decide whether to fetch another page of results. [ts-client#1232](https://github.com/xataio/client-ts/pull/1232)
```

- **Improved Python syntax and error messages:** The playground now offers better syntax highlighting and a more intuitive display of errors for Python code.

> - **Intiating branch movement between cells:** Work has begun on a feature to move branches between cells, facilitating the transfer of databases and data to dedicated clusters. 

Reviewer: ```suggestion
```

we can just remove this one, it's too future facing

> - **Migration to monorepo:** Our site has successfully migrated to a monorepo structure. This means that related Xata projects, components, and resources are now stored in a single repository for improved code management and collaboration. This update lays the foundation for a more personalized and immersive user experience across our app, website, and documentation.

Reviewer: ```suggestion
- **Migration to monorepo:** Our website, application and documentation have successfully migrated to a monorepo structure. This means that related Xata projects, components, and resources are now stored in a single repository for improved code management and collaboration. This update lays the foundation for a more personalized and immersive user experience across the different entities.
```

- **Python code snippets and improved connector information:** We’ve added Python code snippets within the playground. Users can now easily integrate and experiment with Python for an improved development experience. Additionally, we've enhanced the connector info for Drizzle and Kysely snippets across local repositories, covering all data types for a better user experience.
    ![Python code snippet](images/python-code-snippet.png)

> - **Support for importing vector columns:** With this enhancement, you can now import and work with vector columns in your data.

Reviewer: ```suggestion
- **Support for importing vector columns:** With this enhancement, you can now import vector columns through the in-app CSV import wizard.
```

> - **Tracking usage metrics:** Progress has been made in gathering usage metrics, including Postgres disk usage and `ask` endpoint usage, which are now being sent to Clickhouse.

Reviewer: ```suggestion
```

Sorry, not sure if you caught the output of our discussion in slack (it's hard, slack convos are everywhere) but we decided to not highlight this functionality until at least we deliver the initial feature. We can remove this one too
